runner.py

```diff
@@ -124,6 +124,7 @@ def __init__(self,
         self.__start_measurement = None
         self.__end_measurement = None
         self.__services_to_pause_phase = {}
+        self.__join_default_network = False
```

Comment:
Why are you introducing this new variable? Isn't it enough to just check if `networks` is configured in the `usage_scenario` and if not add the default as you do below.

Reply:
setup_networks() is the function to determine if a network key is in the usage_scenario or not.

I could run the same check again later to determine if that is the case or not, but I opted for setting an internal state, so that later functions to not need to know the whys and wheres when a default network  is needed but can rely on a defined internal state. This because I need the information in two places